src/main/java/com/google/sps/servlets/DashboardServlet.java

```diff
@@ -113,4 +111,29 @@ private String convertToJson(HashMap<String, Double> data) {
     String json = gson.toJson(data);
     return json;
   }
+
+ /**
+  * Given the start and end timestamps requested, this function validates and sets 
+  * the instance variables.
+  *
+  * @param startTimestamp A string containing the requested start time as a UNIX timestamp
+  * @param endTimestamp A string containing the requested end time as a UNIX timestamp
+  */
+  private void validateAndSetVariables(String startTimestamp, String endTimestamp){
+
+    if (startTimestamp.isEmpty() || endTimestamp.isEmpty()) {
+      this.startTimestamp = DEFAULT_START_TIMESTAMP;
+      this.endTimestamp = DEFAULT_END_TIMESTAMP;
+    
+    } else {  
+      // swap if they're in the wrong order
```

Comment:
why would they be in the wrong order if you are getting these values from the RequestUtils.getParameter

Reply:
only if the user puts in the dates backwards. decided not to address that in newer commits